### PR TITLE
Ptrus/feature/dependabot

### PR DESCRIPTION
**Pull request by ptrus**
Fixes: #3238 

Configures dependabot for Rust, Go and Github Actions. Also updates all dependencies that were straightforward to update.

After the PR is merged, an admin has to enable dependabot on the repository security settings (enable dependency graph and dependabot alerts).

Notes:
- In cases with long package names (mostly go) dependabot commits won't pass our git lint checks, max commit line length is currently not configurable (https://github.com/dependabot/dependabot-core/issues/2445) - in those cases someone will need to manually update the commit
- for buildkite CI to run automatically for dependabot PRs we will need to add the dependabot email to buildkite organization (https://github.com/buildkite/feedback/issues/313)
- Currently depandabot doesn't tidy the `go.mod` file so we will have to manually update those PRs (https://github.com/dependabot/dependabot-core/issues/2229)
- Added an exception to "changelog fragments check" to allow dependabot submit PRs pass CI without fragments (since this cannot be configured). NOTE: since in any case some dependabot PRs will need manual intervention (see above points) we could also remove this, and always require that someone manually adds a changelog fragment.
- With current configuration (by default) each of the dependabot jobs will open at most 5 PR at a time (others will be queued)
- All dependencies are either updated or ignored, so initially dependabot should not create any PRs

Should probably do a follow-up issue for updating rust dependencies.

### CODE REVIEW

**Review by kostko — COMMENTED** [review]
> for buildkite CI to run automatically for dependabot PRs we will need to add the dependabot email to buildkite organization (buildkite/feedback#313)

I think it's fine if it needs to be started manually.

> Should probably do a follow-up issue for updating rust dependencies.

:+1:

File: runtime/src/common/crypto/signature.rs [hunk 1]
```diff
@@ -3,7 +3,7 @@ use std::io::Cursor;
 
 use anyhow::Result;
 use byteorder::{LittleEndian, ReadBytesExt};
-use ed25519_dalek;
+use ed25519_dalek::{self, ed25519::signature::Signature as _, Signer as _, Verifier};
```
[the remark this answers is not in the document]
Comment (ptrus) [reply runtime/src/common/crypto/signature.rs:6]: as this just imports those traits to be in scope for: `items from traits can only be used if the trait is implemented and in scope` error (and there are already other objects named `Signature` & `Signer`  in the file)

[the remark this answers is not in the document]
Comment (kostko) [reply runtime/src/common/crypto/signature.rs:6]: Oh so because there are other types with the same name, ok.

**Comment by ptrus** [comment]
> I think it's fine if it needs to be started manually.

Ok agreed, and we can always add it later in case we want :+1:

File: .github/workflows/ci-reproducibility.yml [hunk 2]
```diff
@@ -34,7 +34,7 @@ jobs:
           # supported by actions-rs/toolchain.
           cp build${{ matrix.build_number }}/rust-toolchain .
       - name: Set up Go 1.15
-        uses: actions/setup-go@v1.1.2
+        uses: actions/setup-go@v2.1.2.1.2
```
[the remark this answers is not in the document]
Comment (ptrus) [reply .github/workflows/ci-reproducibility.yml]: fixed :-)

**Comment by ptrus** [comment]
> Should probably do a follow-up issue for updating rust dependencies.

I'll open an issue once this is merged, so i can link to the list of ignored dependencies in the `dependabot.yml` :grin:

**Review by kostko — APPROVED** [review]

**Pull request merged**